modules/spatiotemporal/pyext/src/prepare_protein_library.py

```diff
@@ -0,0 +1,200 @@
+           or subcomplexes included in this dictionary will be scored. For
+           each of these proteins, a csv file should be provided with protein
+           copy number data. The csv file should have 3 columns,
+           1) "Time", which matches up to the possible times in the graph,
+           2) "mean", the average protein copy number at that time point
+           from experiment, and 3) "std", the standard deviation of that
+           protein copy number from experiment.
+        @param expected_subcomplexes: list of all possible subcomplex strings
+           in the model. Should be a list without duplicates of
+           all components in the subcomplex configuration files.
+        @param nmodels: int, number of models with different protein copy numbers to generate at each time point.
+        @param output_dir: string, directory where the output will be written.
+           Empty string assumes the current working directory.
+        @param template_topology: string, name of the topology file for the complete complex.
+            (default: '', no topology files are output)
+        @param template_dict: dictionary for connecting the spatiotemporal model to the topology file.
+            The keys (string) are the names of the proteins, defined by the expected_complexes variable.
+            The values (list) are the names of all proteins in the topology file that should have the same copy number
+             as the labeled protein, specifically the "molecule_name."
+            (default: {}, no topology files are output)
+        @param match_final_state: Boolean, determines whether to fix the final state to the state defined by
+            expected_subcomplexes. True enforces this match and thus ensures that the final time has only one state.
+            (default: True)
+        """
+    # Assert that all inputs are the correct variable type
+    if not isinstance(times, list):
+        raise TypeError("times should be of type list")
+    if not isinstance(exp_comp_map, dict):
+        raise TypeError("times should be of type dict")
+    if not isinstance(expected_subcomplexes, list):
+        raise TypeError("nmodels should be of type list")
+    if not isinstance(nmodels, int):
+        raise TypeError("nmodels should be of type int")
+    if not isinstance(output_dir, str):
+        raise TypeError("output_dir should be of type str")
+    if not isinstance(template_topology, str):
+        raise TypeError("template_topology should be of type str")
+    if not isinstance(template_dict, dict):
+        raise TypeError("template_dict should be of type dict")
+    if not isinstance(match_final_state, bool):
+        raise TypeError("match_final_state should be of type bool")
+    # make output_dir if necessary
+    if len(output_dir) > 0:
+        if os.path.exists(output_dir):
+            os.chdir(output_dir)
+        else:
+            os.mkdir(output_dir)
+            os.chdir(output_dir)
```

Comment:
A little inelegant since you `chdir` in both branches. Would be cleaner if the `chdir` were outside the `if`.